inference-engine/src/inference_engine/network_serializer.cpp

```diff
@@ -4,165 +4,276 @@
 
 #include "network_serializer.hpp"
 
-#include <map>
 #include <deque>
+#include <map>
 #include <string>
 #include <vector>
 
-#include "exec_graph_info.hpp"
-#include "xml_parse_utils.h"
 #include <legacy/ie_ngraph_utils.hpp>
-#include <ngraph/variant.hpp>
 #include <ngraph/function.hpp>
+#include <ngraph/ops.hpp>
+#include <ngraph/variant.hpp>
+#include "exec_graph_info.hpp"
+#include "ngraph/opsets/opset.hpp"
+#include "xml_parse_utils.h"
 
 namespace InferenceEngine {
 namespace Serialization {
 
 namespace {
 
-void FillXmlDocWithExecutionNGraph(const InferenceEngine::ICNNNetwork& network,
-                                   pugi::xml_document& doc) {
-    std::shared_ptr<const ngraph::Function> function = network.getFunction();
-    if (function == nullptr) {
-        THROW_IE_EXCEPTION << network.getName() << " does not represent ngraph::Function";
+struct Edge {
+    int from_layer = 0;
+    int from_port = 0;
+    int to_layer = 0;
+    int to_port = 0;
+
+    Edge() = default;
+    Edge(int a, int b, int c, int d)
+        : from_layer{a}, from_port{b}, to_layer{c}, to_port{d} {}
+};
+
+struct ConstantAtributes {
+    int size = 0;
+    int offset = 0;
+};
+
+template <typename T>
+std::string create_atribute_list(
+    ngraph::ValueAccessor<std::vector<T>>& adapter) {
```

Comment:
In this file: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/plugin_api/debug.h we already have helpers which do the same things. You can use their for serialisation

Reply:
Thx for that! We should seriously rename this file to something more meaningful :)